doc/api/http2.md

```diff
@@ -1457,6 +1457,13 @@ numeric stream identifier.
 
 <!-- YAML
 added: v8.4.0
+deprecated:
+  - REPLACEME
```

Comment:
We should probably add an entry to `deprecation.md`, and backport that deprecation to LTS releases

Reply:
Should I send a separate PR for that? Might be easier to backport.

Reply:
Yes, it would be ideal to have a separate PR for the doc-only deprecation, and this PR would move it to EOL assuming a TSC motion supports that plan